### PR TITLE
First pass at parallel_publish

### PR DESCRIPTION
**Pull request by brianbroderick**
This is a naive first pass at parallel publish.

### CODE REVIEW

File: lib/rivulet/kafka/publisher.ex [hunk 1]
```diff
@@ -89,26 +89,33 @@ defmodule Rivulet.Kafka.Publisher do
...
+      tasks
+      |> Enum.zip(messages)
+      |> Map.new
+
+    Task.yield_many(tasks)
     |> Enum.each(fn
-      ({_, {:exit, err}}) ->
-        IO.puts "Republishing due to received errors:"
+      ({task, {:exit, err}}) ->
+        IO.puts "Republishing #{lookup_map[task].key} due to received errors:"
```
Comment (CJPoll) [inline lib/rivulet/kafka/publisher.ex]: `Logger.warn` or `Logger.error` would probably be more appropriate.

File: lib/rivulet/kafka/publisher.ex [hunk 2]
```diff
@@ -89,26 +89,33 @@ defmodule Rivulet.Kafka.Publisher do
...
-      ({_, {:ok, value}}) ->
-        IO.inspect value
-      ({_, nil}) ->
+        publish(lookup_map[task], counter)
+      ({task, {:ok, _}}) ->
+        IO.puts "Published #{lookup_map[task].key}"
+      ({task, nil}) ->
         IO.puts "Received 'nil', republishing"
-        publish(messages, counter)
+        publish(lookup_map[task], counter)
```
Comment (CJPoll) [inline lib/rivulet/kafka/publisher.ex]: The `nil` case means that the task is still executing. Do you think we should attempt a republish, or what a bit longer?

Comment (brianbroderick) [reply lib/rivulet/kafka/publisher.ex]: It depends on how often this happens and how long it normally takes (which I don't know the answer to either).  I'm fine increasing the timeout

File: lib/rivulet/kafka/publisher.ex [hunk 3]
```diff
@@ -61,16 +61,62 @@ defmodule Rivulet.Kafka.Publisher do
...
+
+  @spec handle_task(%{Task.t => [Message.t]}, {Task.t, {:exit, term} | {:ok, term} | nil})
+  :: {:still_running, Task.t} | :ok | :error
+  defp handle_task(_lookup_map, {_task, {:ok, {:error, _}}}), do: :error
+  defp handle_task(_lookup_map, {_task, {:ok, :leader_not_available}}), do: :error
+  defp handle_task(_lookup_map, {_task, {:ok, :ok}}), do: :ok
+  defp handle_task(_lookup_map, {_task, {:ok, _}}), do: :ok
+  defp handle_task(_lookup_map, {task, nil}), do: {:still_running, task}
+  defp handle_task(lookup_map, {task, {:exit, err}}) do
+    Logger.error("Republishing #{lookup_map[task].key} due to received errors: #{inspect err}")
```
Comment (brianbroderick) [inline lib/rivulet/kafka/publisher.ex]: We're returning error, not republishing

**Pull request merged**